related to bug #193258, but this is the final patch that Fedora went with

--- make-3.81/main.c	2007-09-24 15:28:34.000000000 +0200
+++ make-3.81-pm/main.c	2007-09-24 15:32:50.000000000 +0200
@@ -1669,8 +1669,12 @@ main (int argc, char **argv, char **envp
 
     if (job_slots > 0)
       {
-        close (job_fds[0]);
-        close (job_fds[1]);
+	if (restarts == 0)
+	  {
+	    close (job_fds[0]);
+	    close (job_fds[1]);
+	  }
+
         job_fds[0] = job_fds[1] = -1;
         free (jobserver_fds->list);
         free (jobserver_fds);
